packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,87 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+import { filterConfig } from './util.js'
+
+const ALLOWED_PROPERTIES = [
+  ['build', 'edge_functions'],
+  ['build', 'functions'],
+  ['build', 'publish'],
+  ['functions', '*'],
+  ['functions', '*', '*'],
+  ['headers'],
+  ['images', 'remote_images'],
+  ['redirects'],
+]
+
+const OVERRIDE_PROPERTIES = new Set(['headers!', 'redirects!'])
+
+const coreStep: CoreStepFunction = async function ({
+  buildDir,
+  netlifyConfig,
+  systemLog = () => {
+    // no-op
+  },
+}) {
+  const configPath = resolve(buildDir, '.netlify', 'deploy', 'v1', 'config.json')
+
+  let config: Partial<NetlifyConfig> = {}
+
+  try {
+    const data = await fs.readFile(configPath, 'utf8')
+
+    config = JSON.parse(data) as Partial<NetlifyConfig>
+  } catch (err) {
+    // If the file doesn't exist, this is a non-error.
+    if (err.code === 'ENOENT') {
+      return {}
+    }
+
+    systemLog(`Failed to read Deploy Configuration API: ${err.message}`)
+
+    throw new Error('An error occured while processing the platform configurarion defined by your framework.')
```

Comment:
Non-blocker. Should we consider exposing `err.message` here as way for folks to potentially self serve?

Reply:
I'm not sure they can self-serve a solution if the error occurs on a file that is written by their framework, though. 😕